client.go

```diff
@@ -129,28 +130,45 @@ func selectImageProvider(imgStr string, source image.Source, cfg config) (image.
 			return nil, err
 		}
 	case image.OciDirectorySource:
-		if cfg.Platform != nil {
-			return nil, platformSelectionUnsupported
-		}
 		provider = oci.NewProviderFromPath(imgStr, tempDirGenerator)
 	case image.OciTarballSource:
-		if cfg.Platform != nil {
-			return nil, platformSelectionUnsupported
-		}
 		provider = oci.NewProviderFromTarball(imgStr, tempDirGenerator)
 	case image.OciRegistrySource:
 		provider = oci.NewProviderFromRegistry(imgStr, tempDirGenerator, cfg.Registry, cfg.Platform)
 	case image.SingularitySource:
-		if cfg.Platform != nil {
-			return nil, platformSelectionUnsupported
-		}
 		provider = sif.NewProviderFromPath(imgStr, tempDirGenerator)
 	default:
 		return nil, fmt.Errorf("unable to determine image source")
 	}
 	return provider, nil
 }
 
+func setPlatform(source image.Source, cfg *config, defaultArch string) error {
```

Comment:
nit: `setDefaultPlatform` or similar, since it does nothing if platform is set?

Reply:
Well, on second thought, this is also verifying that a non-default, non-nil platform is allowed to be set by the user... which goes beyond setting the default. There is probably a better name for this but setPlatform and setDefaultPlatform both don't quite fit the bill... but setPlatform is slightly less incorrect, so I'll leave it for now.